Feat/restore linux aarch64 builds

mikemhenry opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

mikemhenry committed:
see if we can get aarch64 builds working

mikemhenry committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.27.15.27.40

mikemhenry committed:
bump build number

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-souschef (grayskull)`. This parser is not currently used by conda-forge, but may be in the future. We are collecting information to see which recipes are compatible with grayskull.
 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/21680986178](https://github.com/conda-forge/conda-forge-webservices/actions/runs/21680986178). Examine the logs at this URL for more detail.</sub>

mikemhenry committed:
use older cmake

mikemhenry committed:
MNT: Re-rendered with conda-smithy 3.53.0 and conda-forge-pinning 2025.11.09.23.10.12

Other tools:
- conda-build 25.4.2
- rattler-build 0.16.2
- rattler-build-conda-compat 1.4.6

mikemhenry committed:
MNT: Re-rendered with conda-smithy 3.53.0 and conda-forge-pinning 2025.11.09.23.10.12

Other tools:
- conda-build 25.4.2
- rattler-build 0.16.2
- rattler-build-conda-compat 1.4.6

mikemhenry committed:
rebuild ci support files + add aarch64 builds

mikemhenry committed:
don't install mkl on aarch64 linux

Commit:
np.compat needed by parmed

mikemhenry committed:
bump

mikemhenry committed:
Merge remote-tracking branch 'refs/remotes/origin/feat/restore-linux-aarch64-builds' into feat/restore-linux-aarch64-builds

mikemhenry commented:
@conda-forge-admin, please rerender

mikemhenry commented:
I've gotten local builds to work but the aarch64 builds are timing out, so I need more CI resources.

Commit:
MNT: Re-rendered with conda-smithy 3.54.2 and conda-forge-pinning 2026.02.04.13.18.2

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.24.14.56.5

mikemhenry committed:
Merge branch 'main' into feat/restore-linux-aarch64-builds

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.25.08.09.3

Commit:
fix lint, I think this will double quote things but the linter wants it

Commit:
set github_actions for aarch64 builds

Commit:
use depot runner for aarch64 builds

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.25.08.09.32

Commit:
pin to older boost version

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.25.08.09.32

Commit:
downgrade c/cxx compiler to 13.2.0 for linux

Commit:
maybe they need to be strings

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.25.08.09.32

Commit:
downgrade didn't work, just not going to worry about mpi builds

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.25.08.09.32

jaimergp committed:
Use native MS-hosted runners when possible

MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.06.26.16.14.58

Other tools:
- conda-build 26.5.0
- rattler-build 0.67.0
- rattler-build-conda-compat 1.4.15

Commit:
drop osx x86 builds that are failing

Commit:
see if this fixes osx arm64 builds

Commit:
Fix syntax for dep

jaimergp commented on recipe/meta.yaml:
```diff
@@ -132,8 +134,9 @@ requirements:
-    - mkl  # [linux]
+    - mkl  # [linux and x86_64]
     - __cuda  # [not osx and cuda_compiler_version != "None"]
+    - libpq =18.4=ha770aab_0  # [osx and arm64] 
```
```suggestion
```

jaimergp commented on recipe/meta.yaml:
```diff
@@ -73,12 +75,12 @@ requirements:
     - bzip2
     - zlib
-    - libboost-devel
+    - libboost-devel ==1.86
```
This may become problematic. Why do we need to pin it?

mikemhenry replied:
Code doesn't work with newer versions of boost https://github.com/conda-forge/ambertools-feedstock/pull/181#issuecomment-4803315301

Also working with David Case, ambertools 26 also doesn't work with newer boost yet so there isn't a patch to backport 

>boost is required only for the "moft" component; since this is rarely
     used, and is also available in the AmberClassic components, it could be
     disabled.  A user has sent a proposed patch for the current boost
     library, but I've not had time to test it.
     
Disabling and dropping it would be best which I will try and do for ambertools 25 and 26 but I wanted to try and get this 24 re-build for aarch64 out the door being as close to the ambertools 24 build as possible.

jaimergp commented on recipe/meta.yaml:
```diff
@@ -73,12 +75,12 @@ requirements:
     - python
     - setuptools
-    - numpy
+    - numpy <2.3
```
Same here, why? At the very least, let's add a comment.

mikemhenry replied:
np.compat is needed by parmed BUT it looks like a "new" release https://github.com/ParmEd/ParmEd/releases/tag/4.3.1 may have fixed that.

mikemhenry replied:
I agree that a comment is worth adding but I'll also try removing the pin and seeing what happens

dacase replied:
FWIW, my meta.yaml recipe for ambertools-dac=26 doesn't pin numpy to version 2.  However: (a) I'm unsure if conda-build might still default to version 2; (b) I'm not good enough at python to try to backport whatever changed to earlier versions of AmberTools.

mikemhenry replied:
@dacase That is good to know, in the recipe you sent me, numpy was pinned in the `build` section `- numpy<2` -- I'll remove the pin here and just see if it works (even if parmed is fixed, this could just reveal another incompatibility with `numpy >= 2.3`

dacase replied:
Aargh: my mistake: I as looking at the wrong branch.  My actual recipe for ambertools-dac=26 still has numpy pinned to version 2.  Apologies....

mikemhenry replied:
No worries! it looks like dropping that pin worked for ambertools 24, so I bet it will work for ambertools 25 and 26

jaimergp committed:
Remove libpq pin

jaimergp committed:
Enable CI artifacts

MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.07.07.06.32.36

Other tools:
- conda-build 26.5.0
- rattler-build 0.67.0
- rattler-build-conda-compat 1.4.15

jaimergp committed:
Merge branch 'feat/restore-linux-aarch64-builds' of github.com:mikemhenry/ambertools-feedstock into feat/restore-linux-aarch64-builds

mikemhenry committed:
drop numpy <2.3 pin

mikemhenry commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-smithy 2026.6.14 and conda-forge-pinning 2026.07.07.06.32.3
